core/src/main/java/com/onewelcome/core/usecase/GetAuthenticatedUserProfileUseCase.kt

```diff
@@ -0,0 +1,23 @@
+package com.onewelcome.core.usecase
+
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.onFailure
+import com.github.michaelbull.result.onSuccess
+import com.github.michaelbull.result.runCatching
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onewelcome.core.omisdk.facade.OmiSdkFacade
+import kotlinx.coroutines.suspendCancellableCoroutine
+import javax.inject.Inject
+import kotlin.coroutines.resume
+
+class GetAuthenticatedUserProfileUseCase @Inject constructor(private val omiSdkFacade: OmiSdkFacade) {
+  suspend fun execute(): Result<UserProfile?, Throwable> {
+    return suspendCancellableCoroutine { continuation ->
+      runCatching { omiSdkFacade.oneginiClient.getUserClient().authenticatedUserProfile }
+        .onSuccess { continuation.resume(Ok(it)) }
+        .onFailure { continuation.resume(Err(it)) }
+    }
```

Comment:
coroutine is not needed here, as the SDK method is synchronous. This should suffice
`return runCatching { omiSdkFacade.oneginiClient.getUserClient().authenticatedUserProfile }`